src/org/rascalmpl/library/util/PathConfig.java

```diff
@@ -470,16 +473,23 @@ public static PathConfig fromSourceProjectRascalManifest(ISourceLocation manifes
         IListWriter libsWriter = vf.listWriter();
         IListWriter srcsWriter = vf.listWriter();
         IListWriter messages = vf.listWriter();
-        
-        if (!projectName.equals("rascal")) {
-            // always add the standard library but not for the project named "rascal"
-            // which contains the source of the standard library
-            try {
-                libsWriter.append(resolveCurrentRascalRuntimeJar());
-            }
-            catch (IOException e) {
-                messages.append(Messages.error(e.getMessage(), manifestRoot));
-            }
+
+        // Approach:
+        //   - Rascal modules of `rascal` inside `std` are loaded from the
+        //     "current `rascal`" (as per `resolveCurrentRascalRuntimeJar`).
+        //   - Rascal modules of `rascal` outside `std` are loaded from the
+        //     "current project" of this path config (as per `manifestRoot`).
+        //   - Java classes of `rascal`, regardless of inside/outside `std`, are
+        //     loaded from the current `rascal`.
+        //
+        // Thus, Rascal modules and Java classes of `rascal` inside `std` are
+        // guaranteed to be consistently loaded from the same version.
+        try {
+            srcsWriter.append(URIUtil.rootLocation("std"));
```

Comment:
is there a way to not use `std` ? Or is there no way to figure out what's the rascal.jar that's on the class path?

Reply:
I would be fine with that.